### DISCUSSION TITLE
Question regarding syntax for Il2Cpp.trace()

### DISCUSSION THREAD

**Discussion by 71256988** [vfsfitvnm/frida-il2cpp-bridge#289] [Q&A] [answered]
Hello, I recently found out about this project and was wondering regarding if the syntax for `Il2Cpp.trace()` has changed recently or if I'm doing something wrong.
When I'm trying to build index.ts with the example shown in README.md (+the syntax change mentioned in the 0.8.0 announcement post), I get an error. Similarly, the snippets on the wiki results in the same error.
```javascript
import "frida-il2cpp-bridge";

Il2Cpp.perform(() => {
    Il2Cpp.trace()
        .assemblies(Il2Cpp.domain.assembly("Assembly-CSharp"))
        .and()
        .attach("full");
});
``` 
I get the following error:
`index.ts(5,10): error TS2339: Property 'assemblies' does not exist on type 'Pick<Tracer, "parameters">'.`
I get the same error for `.classes()` as well.
I don't get any error with `Il2Cpp.backtrace()` using the example code shown in the 0.8.0 announcement post.
Was there a change in the recent code rework or am I missing anything?

**Comment by vfsfitvnm** [comment, marked as answer]
I apologize - I forgot to mention that one in the release notes!

After `trace()`, you must call `parameters(true | false)`, where `true` means "print parameters, please" (equivalent to the old `attach("full")`:
```ts
import "frida-il2cpp-bridge";

Il2Cpp.perform(() => {
    Il2Cpp.trace()
        .parameters(true)
        .assemblies(Il2Cpp.domain.assembly("Assembly-CSharp"))
        .and()
        .attach();
});
```

I strongly suggest to use TypeScript w/ a language server (I personally use VSCode):
![Screenshot_20230430_211940](https://user-images.githubusercontent.com/46219656/235372315-8116a669-763e-4242-8dc9-577debfd66f9.png)

**Reply by 71256988** [reply]
Thanks for the quick reply!
